### PR TITLE
feat: add settings page

### PR DESCRIPTION
**Pull request by junlong4321**

### CODE REVIEW

File: frontend/src/pages/AccountSettingPage.tsx [hunk 1]
```diff
@@ -1,31 +1,138 @@
...
-      />
-      <Button variant="contained" onClick={onOpen}>
-        Update Password
-      </Button>
-    </>
+    <Paper elevation={1} sx={{ borderRadius: 3, mx: 6 }}>
+      <Stack direction="row">
+        <List>
+          <ListItem disablePadding sx={{ bgcolor: "white" }}>
+            <ListItemButton
```
Comment (junlong4321) [inline frontend/src/pages/AccountSettingPage.tsx:37]: Is there a way to refactor this part of the code using NavigationButton? I want to replace this part but the onClick action is inside the NavigationButton. I will need to pass down props if I want to reuse the component here

File: frontend/src/pages/AccountSettingPage.tsx [hunk 2]
```diff
@@ -1,31 +1,138 @@
...
+              <PasswordRequirement label="aa" description="Lowercase" />
+            </Stack>
+            {/* To add change display name form here */}
+            <UpdatePasswordForm userId={user?.userId || 0} />
+          </Stack>
+        ) : formType === "changepassword" ? (
+          <Stack spacing={4} sx={{ py: "20px", pl: 6, width: "60%" }}>
+            <Stack spacing={0.5}>
+              <Typography variant="body1" sx={{ fontWeight: "bold" }}>
+                Reset Password
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:113]:
```suggestion
                Update Your Password
```

File: frontend/src/pages/AccountSettingPage.tsx [hunk 3]
```diff
@@ -1,31 +1,138 @@
...
+              </Typography>
+              <Typography variant="body2">
+                Just type it twice and try not to forget it.
+              </Typography>
+              <Typography variant="body2">
+                Password should be and must contain:
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:119]:
```suggestion
                Your password should contain:
```

File: frontend/src/pages/AccountSettingPage.tsx [hunk 4]
```diff
@@ -1,31 +1,138 @@
...
+              <ListItemText primary="Password" />
+            </ListItemButton>
+          </ListItem>
+        </List>
+        <Divider orientation="vertical" flexItem sx={{ mr: "-1px" }} />
+        {formType === "changename" ? (
+          <Stack spacing={4} sx={{ py: "20px", pl: 6, width: "60%" }}>
+            <Stack spacing={0.5}>
+              <Typography variant="body1" sx={{ fontWeight: "bold" }}>
+                Update Display Name
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:90]:
```suggestion
                Update Your Display Name
```

File: frontend/src/pages/AccountSettingPage.tsx [hunk 5]
```diff
@@ -1,31 +1,138 @@
...
+              </Typography>
+              <Typography variant="body2">Personalize Your Name</Typography>
+              <Typography variant="body2">
+                Display Name must contain:
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:94]:
```suggestion
                Your display name should contain:
```

File: frontend/src/pages/AccountSettingPage.tsx [hunk 6]
```diff
@@ -1,31 +1,138 @@
...
+              </Typography>
+            </Stack>
+            <Stack
+              direction="row"
+              sx={{ justifyContent: "center" }}
+              spacing={3}
+            >
+              <PasswordRequirement label="8+" description="Character" />
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:102]: We don't actually have any restrictions on the name, other than that it cannot be empty right now.
```suggestion
              <PasswordRequirement label="3+" description="Character" />
```

File: frontend/src/pages/AccountSettingPage.tsx [hunk 7]
```diff
@@ -1,31 +1,138 @@
...
+                        : "blueGrey.500",
+                  }}
+                />
+              </ListItemIcon>
+              <ListItemText primary="Password" />
+            </ListItemButton>
+          </ListItem>
+        </List>
+        <Divider orientation="vertical" flexItem sx={{ mr: "-1px" }} />
+        {formType === "changename" ? (
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:86]: Can we abstract the two components out into their own files?

File: frontend/src/pages/AccountSettingPage.tsx [hunk 8]
```diff
@@ -1,31 +1,138 @@
...
+                    color:
+                      formType === "changepassword"
+                        ? "primary.500"
+                        : "blueGrey.500",
+                  }}
+                />
+              </ListItemIcon>
+              <ListItemText primary="Password" />
+            </ListItemButton>
+          </ListItem>
```
Comment (dextertanyj) [inline frontend/src/pages/AccountSettingPage.tsx:83]:
```suggestion
          <ListItem disablePadding sx={{ bgcolor: "white" }}>
            <NavigationButton
              buttonDescription={"Display Name"}
              selected={formType === "changename"}
              onClick={(event) => handleListItemClick(event, "changename")}
              Icon={AccountCircle}
              sx={{ pl: 1 }}
            />
          </ListItem>
          <ListItem disablePadding>
            <NavigationButton
              buttonDescription={"Password"}
              selected={formType === "changepassword"}
              onClick={(event) => handleListItemClick(event, "changepassword")}
              Icon={Lock}
              sx={{ pl: 1 }}
            />
          </ListItem>
```

**Review by dextertanyj — APPROVED** [review]
Otherwise, lgtm

**Pull request merged**